pkg/backup/generative_split_and_scatter_processor.go

```diff
@@ -516,7 +524,12 @@ func runGenerativeSplitAndScatter(
 			entry.ProgressIdx = idx
 			idx++
 			if len(chunk.entries) == int(spec.ChunkSize) {
-				chunk.splitKey = entry.Span.Key
+				// We need to split at the start of the next chunk, before passing the
+				// chunk to a chunk worker, to prevent two seperate chunk workers from
+				// accidentally working on the same range, outlined in #146083.
+				if err := baseSplitAndScatterer.split(ctx, flowCtx.Codec(), entry.Span.Key); err != nil {
+					return err
+				}
```

Comment:
I would consider being _painfully_ explicit:

```suggestion
        // The current chunk is full, so this entry will be the start of the next chunk.
        //
				// We split at the start of the next chunk, before passing the full chunk
				// to a chunk scatter worker, to prevent two concurrent chunk workers from
				// working on the same range.
				//
				// See #146083 for more details.
				nextChunkStart := entry.Span.Key
				if err := baseSplitAndScatterer.split(ctx, flowCtx.Codec(), nextChunkStart); err != nil {
					return err
				}
```